v3/src/components/case-table/use-columns.tsx

```diff
@@ -33,6 +33,10 @@ export const useColumns = ({ data, indexColumn }: IUseColumnsProps) => {
       () => {
         const collection = data?.getCollection(collectionId)
         const attrs: IAttribute[] = collection ? getCollectionAttrs(collection, data) : []
+        // if attribute is the only attribute in the collection and it is hidden, make the attribute visible
+        if (attrs.length === 1 && caseMetadata?.isHidden(attrs[0].id)) {
+          caseMetadata?.setIsHidden(attrs[0].id, false)
+        }
```

Comment:
Couple of issues:
- The case table is not the right place to address an issue that doesn't require the case table. (Attributes can be hidden from other components, e.g. the case card.)
- The accessor function of a reaction is just for returning values to the action function, not for actually changing models.
- The number of attributes isn't important. The relevant test is whether all remaining attributes in a collection are hidden, in which case they should all be shown.

I suspect that a reaction that is installed in the `afterCreate()` of the `SharedCaseMetadata` would make the most sense.